torchtune/datasets/_packed.py

```diff
@@ -136,12 +136,15 @@ def _pack(self) -> None:
             # Update the current pack
             current_pack["tokens"] += tokens
             current_pack["labels"] += labels
-            current_pack["input_pos"] += list(range(seq_len))
+            current_pack["input_pos"] += [x % self.max_seq_len for x in range(seq_len)]
```

Comment:
So this is the correct thing to do in terms of fixing the bug, my main question is whether it's the correct thing to do from a model training perspective. I think it is? Basically we are treating the 2nd half of a long sample as a new sample starting from position 0, right? And as long as we are using RoPE (or some kind of positional encoding that only cares about relative position) it doesn't matter that the $(max\textunderscore seq\textunderscore len+1)^{th}$ element now has `input_pos` 0. I guess with absolute positional encoding we should not do this though.

Reply:
I agree with your analysis, and it's unclear to me what choice would be best in all scenarios; some ideas:
1. Disallow any samples that have `seq_len > max_seq_len` (my assumption was that such samples are allowed due to the message of the ValueError on line 131).
2. Truncate the samples that are too long, throw away the rest of the sample.
3. Chunk the samples as done in this PR.
4. Allow either 2 or 3, controlled by a new flag to be added to the `PackedDataset` class, e.g. `truncate_samples` (possibly with truncation being the default behavior). One could also have a warning be emitted in this case if any sample is truncated.

Reply:
Personally I think (3) is most in line with the spirit of `split_across_pack=True` anyways, we don't care too much whether the split is caused by a single really long sample or just awkward spacing of a sample starting right near `max_seq_len` of the pack. For (2) it is reasonable but I actually wouldn't add another config (don't want to make things more complicated than they have to be). One thing we could consider is modifying the behavior of `split_across_pack=False` so that we put any sample whose length exceeds `max_seq_len` in its own pack, truncate it, and raise a warning. I don't have a strong preference between this and the current state of things though.